test/functional/eval/has_spec.lua

```diff
@@ -0,0 +1,51 @@
+local helpers = require('test.functional.helpers')(after_each)
+local eq = helpers.eq
+local clear = helpers.clear
+local funcs = helpers.funcs
+
+describe('has()', function()
+  before_each(clear)
+
+  it('"nvim-x.y.z"', function()
+    eq(0, funcs.has("nvim-"))
+    eq(0, funcs.has("nvim-  "))
+    eq(0, funcs.has("nvim- \t "))
+    eq(0, funcs.has("nvim-0. 1. 1"))
+    eq(0, funcs.has("nvim-0. 1.1"))
+    eq(0, funcs.has("nvim-0.1. 1"))
+    eq(0, funcs.has("nvim-a"))
+    eq(0, funcs.has("nvim-a.b.c"))
+    eq(0, funcs.has("nvim-0.b.c"))
+    eq(0, funcs.has("nvim-0.0.c"))
+    eq(0, funcs.has("nvim-0.b.0"))
+    eq(0, funcs.has("nvim-a.b.0"))
+    eq(0, funcs.has("nvim-0.1"))
```

Comment:
It would be a nice convenience to be able to check for just a major or major.minor version. It'll also fit with principle of least astonishment.